sklearn/manifold/tests/test_spectral_embedding.py

```diff
@@ -19,6 +19,15 @@
 from sklearn.utils._testing import assert_array_almost_equal
 from sklearn.utils._testing import assert_array_equal
 
+try:
+    from pyamg import smoothed_aggregation_solver  # noqa
+
+    is_pyamg_not_available = False
+except ImportError:
+    is_pyamg_not_available = True
+skip_if_no_pyamg = pytest.mark.skipif(
+    is_pyamg_not_available, reason="PyAMG is required for the tests in this function."
+)
```

Comment:
I would prefer to avoid reasoning with multiple negations:

```suggestion
try:
    from pyamg import smoothed_aggregation_solver  # noqa

    pyamg_available = True
except ImportError:
    pyamg_available = False
skip_if_no_pyamg = pytest.mark.skipif(
    not pyamg_available, reason="PyAMG is required for the tests in this function."
)
```

the tests will need to be adapted though.

Reply:
edited directly together with tests